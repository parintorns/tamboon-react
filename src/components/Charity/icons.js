import * as React from 'react'

export function CloseIcon(props) {
  return (
    <svg
      viewBox="0 0 413.348 413.348"
      width={16}
      height={16}
      color="red"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        fill="grey"
        d="M413.348 24.354L388.994 0l-182.32 182.32L24.354 0 0 24.354l182.32 182.32L0 388.994l24.354 24.354 182.32-182.32 182.32 182.32 24.354-24.354-182.32-182.32z"
      />
    </svg>
  )
}

export default CloseIcon
