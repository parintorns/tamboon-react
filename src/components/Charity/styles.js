import styled from 'styled-components'

export const Wrapper = styled.div`
  position: relative;
  margin: 30px;
  width: 500px;
  border-radius: 3px;
  box-shadow: 0px 4px 8px 0px rgba(0, 0, 0, 0.14);
  -webkit-box-shadow: 0px 4px 8px 0px rgba(0, 0, 0, 0.14);
  -moz-box-shadow: 0px 4px 8px 0px rgba(0, 0, 0, 0.14);
`

export const Image = styled.img`
  height: 200px;
  width: 100%;
  object-fit: cover;
`

export const Footer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: baseline;
  margin: 20px;
  justify-content: space-between;
`

export const Title = styled.div`
  font-weight: bold;
`

export const Button = styled.button`
  background-color: #f9f9f9;
  border-radius: 3px;
  border: 1px solid mediumblue;
  display: inline-block;
  cursor: pointer;
  color: mediumblue;
  font-size: 15px;
  font-weight: bold;
  padding: 6px 24px;
  text-decoration: none;
  text-shadow: 0px 1px 0px #ffffff;
  :hover {
    background-color: lightsteelblue;
  }
  :active {
    position: relative;
    top: 1px;
  }
`

export const Modal = styled.div`
  display: ${(props) => (props.visible ? 'flex' : 'none')};
  flex-direction: column;
  position: absolute;
  top: 0px;
  background-color: rgba(255, 255, 255, 0.95);
  width: 100%;
  height: 100%;
`

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 60%;
`

export const ButtonClose = styled.button`
  position: absolute;
  top: 5px;
  right: 5px;
  align-self: flex-end;
  background-color: transparent;
  border-width: 0px;
  cursor: pointer;
  color: #666666;
`

export const Label = styled.div``

export const Option = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
`
