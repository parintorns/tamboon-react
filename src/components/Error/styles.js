import styled from 'styled-components'

export const Wrapper = styled.div`
  position: absolute;
  top: 3px;
  padding: 5px 15px 5px 15px;
  border-radius: 3px;
  background-color: NavajoWhite;
`
