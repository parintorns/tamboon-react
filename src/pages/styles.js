import styled from 'styled-components'

export const Title = styled.h1``

export const Container = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

export const Content = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
`
