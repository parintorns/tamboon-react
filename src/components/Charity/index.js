import React, { useState } from 'react'

import {
  Wrapper,
  Image,
  Footer,
  Title,
  Button,
  Modal,
  ButtonClose,
  Label,
  Body,
  Option
} from './styles'

import { CloseIcon } from './icons'

const payments = [10, 20, 50, 100]

export default function Charity(props) {
  const { charity, onDonate } = props
  const [visible, setVisible] = useState(false)
  const [amount, setAmount] = useState(10)

  const handlePay = () => {
    onDonate(charity.id, amount, charity.currency)

    setVisible(false)
  }

  return (
    <Wrapper>
      <Image src={`/images/${charity.image}`} />

      <Footer>
        <Title>{charity.name}</Title>

        <Button id="btn-donate" onClick={() => setVisible(true)}>
          Donate
        </Button>
      </Footer>

      <Modal visible={visible}>
        <ButtonClose id="btn-close" onClick={() => setVisible(false)}>
          <CloseIcon />
        </ButtonClose>

        <Body>
          <Label>{`Select the amount to donate (${charity.currency})`}</Label>

          <Option>
            {payments.map((payment, index) => (
              <label key={`payment-${charity.id}-option-item-${index}`}>
                <input
                  id={`payment-${charity.id}-option-item-${index}`}
                  type="radio"
                  name="payment"
                  value={payment}
                  checked={amount === payment}
                  onChange={(e) => {
                    setAmount(Number(e.target.value))
                  }}
                />

                {payment}
              </label>
            ))}
          </Option>

          <Button id="btn-pay" onClick={handlePay}>
            Pay
          </Button>
        </Body>
      </Modal>
    </Wrapper>
  )
}
