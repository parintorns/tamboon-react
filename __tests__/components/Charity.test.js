import React from 'react'
import { shallow } from 'enzyme'

import CardDonate from '../../src/components/Charity'
import { Modal } from '../../src/components/Charity/styles'

let wrapper

const props = {
  charity: {
    id: 1,
    name: 'Charity Name',
    currency: 'USD'
  },
  onDonate: jest.fn()
}

beforeEach(() => {
  wrapper = shallow(<CardDonate {...props} />)
})

describe('Render', () => {
  it('should render CardDonate', () => {
    expect(wrapper.debug()).toMatchSnapshot()
  })
})

describe('Actions', () => {
  it('should open donate dialog', () => {
    const btnDonate = wrapper.find({ id: 'btn-donate' })

    expect(btnDonate).toHaveLength(1)

    btnDonate.simulate('click')

    expect(wrapper.find(Modal).prop('visible')).toBe(true)
  })

  it('should close donate dialog', () => {
    const btnClose = wrapper.find({ id: 'btn-close' })

    expect(btnClose).toHaveLength(1)

    btnClose.simulate('click')

    expect(wrapper.find(Modal).prop('visible')).toBe(false)
  })

  it('should select amount', () => {
    const inputAmount = wrapper.find({ id: 'payment-1-option-item-0' })

    expect(inputAmount).toHaveLength(1)

    inputAmount.simulate('click', { target: { value: 10 } })

    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('should make payment', () => {
    const btnPay = wrapper.find({ id: 'btn-pay' })

    expect(btnPay).toHaveLength(1)

    btnPay.simulate('click')

    expect(props.onDonate).toHaveBeenCalledWith(1, 10, 'USD')
  })
})
