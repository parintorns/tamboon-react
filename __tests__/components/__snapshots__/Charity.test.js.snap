// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should select amount 1`] = `
"<styled.div>
  <styled.img src=\\"/images/undefined\\" />
  <styled.div>
    <styled.div>
      Charity Name
    </styled.div>
    <styled.button id=\\"btn-donate\\" onClick={[Function: onClick]}>
      Donate
    </styled.button>
  </styled.div>
  <styled.div visible={false}>
    <styled.button id=\\"btn-close\\" onClick={[Function: onClick]}>
      X
    </styled.button>
    <styled.div>
      <styled.div>
        Select the amount to donate (USD)
      </styled.div>
      <styled.div>
        <label>
          <input id=\\"amount-item-0\\" type=\\"radio\\" name=\\"payment\\" value={10} onClick={[Function: onClick]} />
          10
        </label>
        <label>
          <input id=\\"amount-item-1\\" type=\\"radio\\" name=\\"payment\\" value={20} onClick={[Function: onClick]} />
          20
        </label>
        <label>
          <input id=\\"amount-item-2\\" type=\\"radio\\" name=\\"payment\\" value={50} onClick={[Function: onClick]} />
          50
        </label>
        <label>
          <input id=\\"amount-item-3\\" type=\\"radio\\" name=\\"payment\\" value={100} onClick={[Function: onClick]} />
          100
        </label>
      </styled.div>
      <styled.button id=\\"btn-pay\\" onClick={[Function: handlePay]}>
        Pay
      </styled.button>
    </styled.div>
  </styled.div>
</styled.div>"
`;

exports[`Render should render CardDonate 1`] = `
"<styled.div>
  <styled.img src=\\"/images/undefined\\" />
  <styled.div>
    <styled.div>
      Charity Name
    </styled.div>
    <styled.button id=\\"btn-donate\\" onClick={[Function: onClick]}>
      Donate
    </styled.button>
  </styled.div>
  <styled.div visible={false}>
    <styled.button id=\\"btn-close\\" onClick={[Function: onClick]}>
      X
    </styled.button>
    <styled.div>
      <styled.div>
        Select the amount to donate (USD)
      </styled.div>
      <styled.div>
        <label>
          <input id=\\"amount-item-0\\" type=\\"radio\\" name=\\"payment\\" value={10} onClick={[Function: onClick]} />
          10
        </label>
        <label>
          <input id=\\"amount-item-1\\" type=\\"radio\\" name=\\"payment\\" value={20} onClick={[Function: onClick]} />
          20
        </label>
        <label>
          <input id=\\"amount-item-2\\" type=\\"radio\\" name=\\"payment\\" value={50} onClick={[Function: onClick]} />
          50
        </label>
        <label>
          <input id=\\"amount-item-3\\" type=\\"radio\\" name=\\"payment\\" value={100} onClick={[Function: onClick]} />
          100
        </label>
      </styled.div>
      <styled.button id=\\"btn-pay\\" onClick={[Function: handlePay]}>
        Pay
      </styled.button>
    </styled.div>
  </styled.div>
</styled.div>"
`;
